server/portal_server.js

```diff
@@ -136,20 +137,24 @@ app.post('/delete/:id', (req, res) => {
 });
 
 app.post('/upload/:id', (req, res, next) => {
```

Comment:
We no longer need the `:id` parameter in the upload url. `new_id` should be created at this level instead of in storage.js. It'd also be better to create `meta.delete` up here too. Ideally storage.js should only be concerned with saving and fetching the data from persistent storage and leaving the logic of what the data is and how it's created to something else (this function).